### ISSUE TITLE
[ENH] add possibility to compute design efficiency for a design matrix and a given contrast (or set of contrasts)

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4526]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed enhancement in detail.

Another "model evaluation" idea: design efficiency can be computed before any data is acquired.

It could be useful to have a function to compute it for a given contrast or set of contrasts.

For more information:

https://www.youtube.com/watch?v=FD4ztsoYvSY&list=PLB2iAtgpI4YEnBdb_jDGmMcdGoIBwhCCY

https://imaging.mrc-cbu.cam.ac.uk/imaging/DesignEfficiency

### Benefits to the change

Allow users to evaluate their model before running experiment

**Comment by Remi-Gau** [comment]
This may be out of scope for Nilearn, but the other package that I am aware of that can help compute this (https://github.com/neuropower/neurodesign) is not actively maintained.

**Comment by bthirion** [comment]
I have the feeling that people don't use much these features, so the danger is to support another functionality that is not used.
But I agree that the cost is minor.

**Comment by Remi-Gau** [comment]
Agreed that people don't compute design efficiency even though they should.
No rush in adding those features.

Could go in the sandbox: https://github.com/nilearn/nilearn_sandbox

**Comment by bthirion** [comment]
-1 on sandbox, which can only lead to a repo of broken code.

**Comment by KTZ228** [comment]
I do think that it would be useful to not have to do this manually.
Making a design matrix is one of the first things users try out when they consider switching from FSL, SPM etc. and missing a somewhat basic feature like this might be offputting for some.
